tests/binary_agreement.rs

```diff
@@ -1,108 +1,137 @@
 #![deny(unused_must_use)]
-//! Tests of the Binary Agreement protocol. Only one proposer instance
-//! is tested. Each of the nodes in the simulated network run only one instance
-//! of Binary Agreement. This way we only test correctness of the protocol and not
-//! message dispatch between multiple proposers.
+//! Tests of the Binary Agreement protocol
+//!
+//! Each of the nodes in the simulated network runs only one instance of Binary Agreement for node
+//! `0`, the proposer. In real applications there would be multiple proposers in the
```

Comment:
I don't think that's generally the case: In _Honey Badger_, binary agreement happens to be about whether a particular proposer's contribution is accepted, but there may well be other uses. E.g. there are consensus algorithms where (something similar to) BA is used to determine whether a particular _transaction_ gets included in the next batch or not. Or imagine someone extends Honey Badger to make it "finite": Each epoch, the validators run a single additional BA instance, and if that return `true`, they all terminate.
So I'd vote for not mentioning the concept of a "proposer" here at all.